conans/search/search.py

```diff
@@ -118,35 +134,34 @@ def search_packages(cache, ref, query):
                            settings: {os: Windows}}}
     param ref: ConanFileReference object
     """
-    if not os.path.exists(cache.conan(ref)):
-        raise NotFoundException("Recipe not found: %s" % str(ref))
+    if not os.path.exists(cache.conan(ref)) or (
+           ref.revision and cache.package_layout(ref).recipe_revision()[0] != ref.revision):
```

Comment:
``cache.package_layout(ref).recipe_revision()[0]``? Why the ``[0]``?

Reply:
It returns a tuple of (revision, revision_time)